### PR TITLE
SAMMIs - Stationary Assembly MMIs

### PR DESCRIPTION
**Pull request by ghost**
sc

Can be made from MMIs with or without brain.
They are made with 2 robot arms, powercell, and an MMI, then finished with a wrench.
Ghosts can freely enter or leave SAMMIs as often as they want.
SAMMIs can be pulled around and anchored with a wrench.
SAMMIs can speak now.
SAMMIs' primary law can now be reprogrammed.
SAMMI sprites show if the unit is active.(Is there a player inside?)

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Could we get some more information on these?
Are they capable of any interaction a human is?
Do they have laws?
Can they be EMP'd?

**Comment by ghost** [comment]
they are less expensive stationary mommis with the law:
"You must work together with the rest of the SAMMI network to complete all tasks given to you"

they can't move, they can be anchored

EDIT: Laws have been changed to:
        "You must follow all laws.",
        "You do not yet have a second law.",
        "Do not allow harm to any sentient being unless that would make following Law 2 impossible."

You can edit the second law

**Comment by Shadowmech88** [comment]
Can they speak?
Do they have their own comms channel?

**Comment by ghost** [comment]
They can't speak.
They are on damage control.

I thought about setting them apart from mommi communication but they could potentially work together.

Also the first (or only) SAMMI built in the round should have someone to chat with i think

EDIT:
They Can now speak

**Comment by ghost** [comment]
Also they are built same as a mommi except:
only need 2 arms, power cell, and mmi
they are finished with a wrench instead of a screwdriver

EDIT:
SAMMIs can also be freely entered and exited by ghosts

**Comment by Shadowmech88** [comment]
Since they can't move, which also means they can't really do their own projects or anything besides specifically what they were built for, I don't think it'd be very fun for the SAMMI player to also not be able to talk with the humans they're intended to serve.

**Comment by carshalash** [comment]
They do still have :b right?

**Comment by NigglyWiggly** [comment]
Does this mean we can have borgs dragging around a set of arms

**Comment by 9600bauds** [comment]
I thought this was going to be some sort of stationary platform that a borg can walk on top of and gain a set of arms for only that tile

I'm a bit disappointed

**Comment by Senaat1** [comment]
I don't see the incentive to ever want to play this. You're like a less useful and fun to play MoMMI that just sits there and is people's bitch.

**Comment by SerWind** [comment]
This is actually really cool but I'd recommend players be able to drop in and out of them at will instead of having them be something some is tied to then they die or go braindead, etc. Also they should definitely be able to communicate with humans. As was mentioned, it would be boring otherwise, and communication will help them do their job.  If you are afraid of people dropping into them for meme supremes just give them a law that says they can't waste time on bullshit.

**Comment by icantthinkofanameritenow** [comment]
Just for the record, this is from the same man who gave /vg/station cancer. 

Make what you will of that.

**Comment by SangriaMonkey** [comment]
This is just fucking bizarre and I can't imagine that anyone would willingly play one of these if they can't even communicate with people.

**Comment by chatsonf** [comment]
Kill me now

**Comment by ghost** [comment]
Clearly they should communicate with people

**Comment by SerWind** [comment]
Furbeh why is your image so fucking spoopy. 

Cloro what are your thoughts and or feelings on them being able to be controlled and abandoned by ghosts on a whim?

**Comment by Praecordia** [comment]
so, it's like a punishment job?
given that only the legendary _hyper-autist_ will ever willingly play a job that can't talk, can't move, and can only really interact with one thing
maybe give ghosts an option to drop into them/leave them whenever

**Comment by chatsonf** [comment]
Just my go-to avatar SerWind.

Don't really like the idea of being controlled and abandoned by ghosts, there's no real incentive for ghosts to bother controlling them to do menial tasks. 
Seems almost torturous to be a SAMMI, like it's worse than being a MoMMI, which is already hell for the standard spess player.

**Comment by AutismSpider** [comment]
@cloroxygen  Give them a edited pos brain law set about helping the crew give them the power to move at slow speeds (human in hard suit speeds)

**Comment by AutismSpider** [comment]
If any thing for access work like mmi in a mech were you have to set it with a id so can have one in medbay doing Surgery etc

**Comment by SerWind** [comment]
I don't see the big deal and I think making them move is a bad idea because it defeats the point. Having players be able to drop in and out of them solves that problem. If there's multiples around the station I guarantee at least one autismo will be bouncing around helping people out, and I think it's a stretch to say ghosts/observers wouldn't enjoy it to some stretch. Less commitment than a regular borg/pAI/mommi, and observers could drop in for ten minutes or whatever without surrendering the ability to go back to observing. 

Meta gaming is obviously a concern but not having good things because of shitters seems more like a punishment for us.

**Comment by clusterfack** [comment]
I've always wanted to roleplay as the robot from Iron Man

**Comment by Senaat1** [comment]
Why not have a SAMMI Core, which is similar to an AI core and give it the ability to jump between inactive SAMMIs at will?

**Comment by kol1th** [comment]
> Why not have a SAMMI Core, which is similar to an AI core and give it the ability to jump between inactive SAMMIs at will?

this THIS i would play that a shit ton

**Comment by SerWind** [comment]
Feels like we are approaching "just have the AI be able to control the sammis" territory.

**Comment by Shadowmech88** [comment]
The SAMMI core wouldn't necessarily have the powers of the AI core though.
It could cycle through the SAMMIs, but wouldn't have the free eye movement that an AI does, meaning it couldn't exert influence over any area that didn't have a SAMMI in it.

If we go with the SAMMI core idea though, should we make it so that SAMMIs must specifically be bound to a core, or should any SAMMI core be able to jump into any empty SAMMI, meaning all the active cores share all the existing SAMMIs?

**Comment by Senaat1** [comment]
That honestly works as well. Only issue would be people law 2ing the AI into helping them do secondary shit while there are other, more pressing matters the AI could be attending to.

**Comment by clusterfack** [comment]
I think these ideas have some momentum

**Comment by AutismSpider** [comment]
Why not make this a upgrade for cyborgs?

Borg gives up two tool slots and some run speed for the power grab stuff like a mommi?

**Comment by SerWind** [comment]
HELP AI ROGUE NOT RESPONDING TO SAMMI REQUESTS

I don't think either idea is bad, for what it's worth. the AI being able to control them or cores being made that have limited control. I am still pretty partial to the idea of ghosts being able to inhabit them as they please though.

**Comment by Senaat1** [comment]
Actually, why not give borgs an ability to deploy into a stationary arm mode that lets them interface with things, but not move. I 'unno.

**Comment by Shadowmech88** [comment]
What major balance concerns would there be in giving the AI access to a hand?
Keep in mind that if there's one SAMMI _anywhere_ on the station, a borg can unwrench it and take it wherever the AI wants it to.

**Comment by SerWind** [comment]
Major balance concerns: AI choking the fuck out of the RD at round start.

**Comment by kol1th** [comment]
dont give those disgusting milfs more tools to fuck people over with

just make a separate ""AI"" core that can only control the SAMMIs

**Comment by Senaat1** [comment]
That sounds great to me, Ser.

**Comment by AutismSpider** [comment]
@Shadowmech88 Make the hand work like the mommis cant use guns or two hand stuff or grab law boards etc

**Comment by Shadowmech88** [comment]
I vote for the separate SAMMI core just because I think AIs have enough on their plate already.

**Comment by SerWind** [comment]
Right? I think AIs should be able to have access to them, but through an upgrade, or some change to the AI core or something. Have them be predominantly a separate entity, but also have the option available. Also I vote we have all three and see what works. AI control, core control, ghost control. See what sticks, why not.

**Comment by Senaat1** [comment]
Okay, alternative options aside, can we all agree a SAMMI with one person bound to it that can't leave it would be horrifyingly boring?

**Comment by ihadtoregisterforthis** [comment]
Yes

**Comment by SerWind** [comment]
Yeah. Would be DoA

**Comment by AutismSpider** [comment]
@Senaat1 Yes

It should be some type of upgrade you glue to a borg or a tool the Ai can use by linking up with it in some silly way

**Comment by Senaat1** [comment]
what if you could combine two SAMMIs and some mech parts to make a mech the ai could inhabit.

**Comment by AutismSpider** [comment]
@Senaat1  long arm piss off Ai chokes the condom to death in his room

**Comment by Senaat1** [comment]
i feel like the AI getting the ability to choke a bitch out is a good change, not a bad one.

**Comment by SerWind** [comment]
I actually think that would be great for a milf AI just for the screams of HELP AI CHOKING ME IN DORMS

But really I think giving borgs an arm upgrade of some sort is a cool idea but does it have to be one or the other? I want robot arms on walls for ultra lewd mechano ERP.

**Comment by AutismSpider** [comment]
@SerWind if the borg ops for arm upgrade it should cost them a tool slot or two or hit in run speed or mix of both

**Comment by ghost** [comment]
At this moment it functions as a pai that can directly interact with the world. Speech has been added, it can also speak over radio.

**Comment by SerWind** [comment]
One of these needs to start in chemistry, if you aren't already planning to put them everywhere.

**Comment by AutismSpider** [comment]
Feels like we can do more with this.

How about a way for the crew to carry it on his or hers back (this should slow you the fuck down big ass metal claw and stuff)

**Comment by SerWind** [comment]
I serve the machine spirit

That's a cool idea too. Hey Cloro, do it all :P

**Comment by AutismSpider** [comment]
it be funny  big talking metal claw to help with work and to talk with and other stuff if you happen to be antag etc

**Comment by ghost** [comment]
right now you can drag it around and anchor it where you want.

and the other features you guys have suggested (being able to jump around in the sammi network and control different machines) is the direction this should go

**Comment by Shadowmech88** [comment]
I like the current incarnation.
However, there should be a very obvious sprite difference between an empty SAMMI and a SAMMI with a ghost in it, so that people don't end up delegating orders to empty ones.
A human interacting with an empty SAMMI should also let ghosts know that someone is requesting a ghost, and provide a link to directly possess the SAMMI.

You should also make sure that a ghost possessing a SAMMI does not affect the ghost's ability to return to its _original_ body.

**Comment by Intigracy** [comment]
> 54 comments in 4 hours

wew lad

**Comment by ghost** [comment]
> You should also make sure that a ghost possessing a SAMMI does not affect the ghost's ability to return to its original body.
> 
> Die, get dragged to the cloner
> Geneticist says "If you want back in, clone yourself"

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 1]
```diff
@@ -82,13 +88,73 @@ obj/item/device/mmi/Destroy()
...
+		if(!builtempty)
+			if(!brainmob.key)
+				if(!mind_can_reenter(brainmob.mind))
+					to_chat(user, "<span class='notice'>\The [src] indicates that their mind is completely unresponsive; there's no point.</span>")
+					return TRUE
+			if(brainmob.stat == DEAD)
+				to_chat(user, "<span class='warning'>Yeah, good idea. Give something deader than the pizza in your fridge legs.  Mom would be so proud.</span>")
+				return TRUE
+			if(brainmob.mind in ticker.mode.head_revolutionaries)
+				to_chat(user, "<span class='warning'>The [src]'s firmware lets out a shrill sound, and flashes 'Abnormal Memory Engram'. It refuses to accept the brain.</span>")
```
Comment (ghost) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Odd check, how come?

**Comment by ghost** [comment]
If the SAMMI is supposed to do a certain task, and other ghosts can quickly swap in and out easily, would it be a good idea to have it so that people can set the task given to the SAMMI, and that's kept as a log that can be read and referred to by any new ghosts that take control?

**Comment by Shadowmech88** [comment]
Actually yeah. Make a verb that can be used to set the SAMMI's objective, then relay the objective to any ghost that joins, like when a silicon sees its laws upon being built.

**Comment by DeityLink** [comment]
I hope you made it so that it can be destroyed through attackby/bullet_act() etc.

I've already got plans to improve its visuals once this PR gets merged.

**Comment by ghost** [comment]
@DeityLink Gonna add it so it supports in-hand or something?

First thing I thought when I saw the hand

**Comment by DeityLink** [comment]
I'll add an animation when it picks up/drops an item, and have the item it's holding appear in miniature, just like with rechargers. Oh and I'll add the sprite a bit of shading around the edges. And maybe another animated sprite for when a ghost is in control.

**Comment by ghost** [comment]
Here's a thing, can they ride wheelchairs?

> Drag self onto wheelchair
> Grab nearby fire extinguisher
> Ride to victory

**Comment by DeityLink** [comment]
It has to be wrenched to work, so no.

**Comment by MercuryDoll** [comment]
This looks nice, gives people who got antagged early round and observers something to do while they're waiting on that 14:30 shuttle.
Question though, if you're going for the SAMMI network, will only a single ghost be able to use it, or can multiple ghosts jump into the network and assume direct control if multiple SAMMI stations are build?
Also, drop me the sprites and i'll try to make them fit the theme of our current R&D machines.

**Comment by Senaat1** [comment]
If the SAMMI is going to be something any ghost can enter at any time, maybe they SHOULDN'T be able to talk? Let them keep deadchat instead of talking to humans because boy this sounds like a great way to meta.

**Comment by Shadowmech88** [comment]
> because boy this sounds like a great way to meta

Then ban the people that use it to meta.
Ghosts can jump into a posibrain and meta that way as well, but that's not cause to remove posibrains' ability to speak.
In fact, I believe the living can see LOOC messages from ghosts. The ghosts don't even have to _be_ in a body in order to meta.

**Comment by Senaat1** [comment]
Yeah, that's true.

File: code/modules/mob/living/carbon/brain/MMI.dm (hunk 1 above)
Comment (gbasood) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Yeah, actually, why IS this check here? If they 're dead, does it matter if their mind is a head rev?

**Comment by gbasood** [comment]
Looks like you branched off of a bugfix branch when you made this
Not a huge issue as far as I'm concerned but it's a bit messy so normally you'd want to avoid that obviously

Although actually those merge commits are stuff you'd DEFINITELY not want in a PR almost all the time

**Comment by Shadowmech88** [comment]
Good thing Github has auto-squashing now.

**Comment by MercuryDoll** [comment]
Honestly i wouldn't mind if SAMMIs couldn't speak with humans, or even on binary and just kept deadchat. As ghost you can still hear mobs around you and you can keep the discussion with other fellow ghosts.

**Comment by Hoshi-chan** [comment]
Ah, yes, the dream of talking toaster borgs will be closer than ever with the implementation of this.

**Comment by FudgePucker** [comment]
Yea it might get more use if you could keep ded chat.
Also put a unwrenched one in chemistry so it can make chems, mix chems, load people into cyro, or manage the cloner.

File: code/modules/mob/living/carbon/brain/MMI.dm (hunk 1 above)
Comment (ghost) [inline code/modules/mob/living/carbon/brain/MMI.dm]: That is just MMI code, look at line 58|64
...I was just not going to make a bunch of unsolicited, loosely associated changes when less alteration accomplishes what i am setting out to do.

I don't know if that is best practice for you guys but it feels like it to me.

**Comment by jknpj** [comment]
Why not make it a docking station? Dock a MMI, posibrain or pAI and make it do your job.

**Comment by gbasood** [comment]
@Shadowmech88 It still fucks up the commit history when you have merge commits inside a PR

**Comment by Ephx1** [comment]
NOICE
More silicon roles are great, even better if they aren't tied down by the retardation of law arguments.

I see a lot of people talking about movement, I think they're fine as is, but if they were to move I'd suggest having them tied to an APC, they move out of the room they instantly shut off. Even still, they should be walking speed slow.

**Comment by Brolaire** [comment]
Can it aflap?
Can it wear hats?
Can it harm humies without breaking laws?

**Comment by PJB3005** [comment]
What the fuck why am I assigned to this and why did Ririchiyo DNM it without reason.

**Comment by ghost** [comment]
Because Riri's a nerd sometimes?

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 2]
```diff
@@ -82,13 +88,73 @@ obj/item/device/mmi/Destroy()
...
+				M.mind.store_memory("In case you look at this after being borged, the objectives are only here until I find a way to make them not show up for you, as I can't simply delete them without screwing up round-end reporting. --NeoFite")
+
+			M.job = "SAMMI"
+
+			M.cell = locate(/obj/item/weapon/cell) in contents
+			M.cell.loc = M
+			src.loc = M//Should fix cybros run time erroring when blown up. It got deleted before, along with the frame.
+			M.mmi = src
+			return TRUE
+		else:
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/brain/MMI.dm]: How the fuck did this compile.

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 3]
```diff
@@ -82,13 +88,73 @@ obj/item/device/mmi/Destroy()
...
+			brainmob.mind.transfer_to(M)
+
+			if(M.mind && M.mind.special_role)
+				M.mind.store_memory("In case you look at this after being borged, the objectives are only here until I find a way to make them not show up for you, as I can't simply delete them without screwing up round-end reporting. --NeoFite")
+
+			M.job = "SAMMI"
+
+			M.cell = locate(/obj/item/weapon/cell) in contents
+			M.cell.loc = M
+			src.loc = M//Should fix cybros run time erroring when blown up. It got deleted before, along with the frame.
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/brain/MMI.dm:145]: `forceMove(M)`

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 4]
```diff
@@ -36,6 +36,157 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
+	prefix = "Stationary Assembler MMI"
+	canmove = 0
+	..()
+
+
+/mob/living/silicon/robot/mommi/sammi/update_canmove()
+	return 0
+
+/mob/living/silicon/robot/mommi/sammi/attackby(obj/item/W, mob/user)
+
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm:71]: Not 1 but 2 lines of whitespace? That's a new one.

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 5]
```diff
@@ -36,6 +36,157 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
+
+	if(istype(W, /obj/item/stack/cable_coil) && wiresexposed)
+		var/obj/item/stack/cable_coil/coil = W
+		adjustFireLoss(-30)
+		updatehealth()
+		coil.use(1)
+		for(var/mob/O in viewers(user, null))
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm]: Visible message.

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 6]
```diff
@@ -36,6 +36,157 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
+	icon = 'icons/mob/robots.dmi'
+	icon_state = "sammi"
+	maxHealth = 60
+	health = 60
+	keeper=0 // 0 = No, 1 = Yes (Disables speech and common radio.)
+	picked = 1
+	subtype="sammi"
+	prefix = "Stationary Assembler MMI"
+	canmove = 0
+	..()
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm]: How the fuck did this compile putting a fucking `..()` in the type definition.

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 7]
```diff
@@ -36,6 +36,157 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
+			if(locked)
+				to_chat(user, "The cover is locked and cannot be opened.")
+			else
+				to_chat(user, "You open the cover.")
+				opened = 1
+				updateicon()
+
+	else if (istype(W, /obj/item/weapon/cell) && opened)	// trying to put a cell inside
+		if(wiresexposed)
+			to_chat(user, "Close the panel first.")
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm]: Should say wiring panel, they might assume cover.

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 8]
```diff
@@ -36,6 +36,157 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
+	subtype="sammi"
+	prefix = "Stationary Assembler MMI"
+	canmove = 0
+	..()
+
+
+/mob/living/silicon/robot/mommi/sammi/update_canmove()
+	return 0
+
+/mob/living/silicon/robot/mommi/sammi/attackby(obj/item/W, mob/user)
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm:70]: Almost this entire proc is copypasta jesus.

File: code/modules/mob/living/silicon/mommi/mommi.dm [hunk 9]
```diff
@@ -96,6 +247,44 @@ They can only use one tool at a time, they can't choose modules, and they have 1
...
 		to_chat(world, "<span class='warning'>ASSERT FAILURE: connected_ai && keeper in mommi.dm</span>")
 
+/mob/living/silicon/robot/mommi/sammi/New(loc)
+	..()
+	module = new /obj/item/weapon/robot_module/mommi/sammi(src)
+	laws = new sammi_base_law_type
+
+
+/mob/living/silicon/robot/mommi/sammi/proc/transfer_personality(var/client/candidate)
+
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi.dm:273]: The whitespace.

File: code/modules/mob/living/silicon/mommi/mommi_modules.dm [hunk 10]
```diff
@@ -35,6 +34,42 @@
...
+	src.modules += new /obj/item/weapon/wirecutters(src)
+	src.modules += new /obj/item/device/multitool(src)
+	//src.modules += new /obj/item/device/t_scanner(src)
+	//src.modules += new /obj/item/device/analyzer(src)
+	src.modules += new /obj/item/weapon/extinguisher(src) // Aurx sed so
+	src.modules += new /obj/item/weapon/extinguisher/foam(src)
+	//src.modules += new /obj/item/device/rcd/rpd(src)
+	//src.modules += new /obj/item/device/rcd/tile_painter(src)
+	//src.modules += new /obj/item/blueprints/mommiprints(src)
+	src.modules += new /obj/item/device/material_synth/robot/mommi(src)
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi_modules.dm:64]: Okay just saying but a mat synth will be abused REALLY BADLY.

File: code/modules/mob/living/silicon/mommi/mommi_modules.dm [hunk 11]
```diff
@@ -35,6 +34,42 @@
...
+	//src.modules += new /obj/item/device/holomap(src)
+	sensor_augs = list("Mesons", "Disable")
+	var/obj/item/stack/cable_coil/W = new /obj/item/stack/cable_coil(src)
+	W.amount = 50
+	W.max_amount = 50 // Override MAXCOIL
+	src.modules += W
+	return
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi_modules.dm]: A return

That trails!

Aaaaaah

File: code/datums/ai_laws.dm [hunk 12]
```diff
@@ -1,7 +1,7 @@
 var/global/randomize_laws      = 0 // Not right now - N3X
 var/global/base_law_type       = /datum/ai_laws/asimov  //Deinitialize this variable by commenting out Asimov as the base_law_type to activate AI lawset randomization
 var/global/mommi_base_law_type = /datum/ai_laws/keeper // Asimov is OP as fuck on MoMMIs. - N3X
-
+var/global/sammi_base_law_type = /datum/ai_laws/sammi
```
Comment (PJB3005) [inline code/datums/ai_laws.dm]: `/global` does not do anything in this context.

File: code/modules/mob/living/silicon/mommi/mommi_modules.dm (hunk 10 above)
Comment (ghost) [inline code/modules/mob/living/silicon/mommi/mommi_modules.dm:64]: this potential exists elsewhere, in a discussion that was had on the subject, it was concluded this was not an issue.

Comment (Intigracy) [inline code/modules/mob/living/silicon/mommi/mommi_modules.dm:64]:
```
[21:25:41] <Sood> Cloroxygen, that issue on possible fabrication abuse will need to be debated out
[21:25:50] <Sood> also yeah
[21:25:53] <Sood> how are sammis powered
[21:25:55] <Sood> I dont remember if oyu answered
[21:26:24] <Cloroxygen> battery only, the mat fabricator uses a ton of energy, i dont think that will be an issue
[21:26:50] <Sood> hm
[21:27:17] <Intigracy> it'll be just about as abusable as borgs doing it
[21:27:24] <Intigracy> so not really that big of a deal imho
[21:27:25] <Sood> well leave that as a reply to pjb's line comment
[21:27:27] <Sood> Intigracy, 
[21:27:37] <Sood> the difference is that sammi could fab plasteel
[21:27:40] <Sood> and hand it out
[21:27:42] <Sood> I think
[21:27:47] <Intigracy> borgs can too
[21:27:57] <Sood> hm
[21:27:57] <Intigracy> if you scan plasteel then generate 50 of it
[21:27:58] <Sood> oh well
[21:28:00] <Intigracy> it puts it in your modules
[21:28:05] <Intigracy> any more generated is dropped on the floor
[21:28:07] <Intigracy> mommi too
[21:28:10] <Sood> welp
[21:28:16] <Sood> Cloroxygen, include that then
[21:28:29] <Intigracy> It's default behavior of the borg mat synth iirc
```

**Comment by AutismSpider** [comment]
@PJB3005 two mat synth mommi and cyborg/normal one mommi one can't copy shit (people hate fun and hate mommis turning the floors and walls into  Bananium)  and has list of mats it can make borg/normal one can copy any thing it wants form wood to plastic

**Comment by ghost** [comment]
this should be ready to merge unless there are any issues to raise?

**Comment by AutismSpider** [comment]
Got update the wiki on how to build one and shit

**Comment by gbasood** [comment]
Before the merge we should get a changelog file in here.
The wiki can wait later, since it usually does anyway
Look at https://github.com/D3athrow/vgstation13/tree/Bleeding-Edge/html/changelogs/example.yml

**Comment by clusterfack** [comment]
Yep besides a changelog this is good to go

**Comment by ghost** [comment]
Anything else?

File: html/changelogs/Cloroxygen.yml [hunk 13]
```diff
@@ -0,0 +1,37 @@
+################################
```
Comment (ghost) [inline html/changelogs/Cloroxygen.yml]: I don't even...give me a sec

**Comment by Brolaire** [comment]
make them wesr hats and sflap

**Comment by ghost** [comment]
@Brolaire 
They already can

**Comment by PJB3005** [comment]
But why?

**Comment by Ephx1** [comment]
Cancer got to him

**Comment by ghost** [comment]
What, but why?

Why!

**Comment by Intigracy** [comment]
He's impatient for merging

**Comment by Brolaire** [comment]
blameriri

**Comment by ghost** [comment]
But, then you just move on to something else, stack up those features, squish those bugs, leave things to stew in the meantime until they're practically fermenting

**Comment by SangriaMonkey** [comment]
_rip_

**Comment by AutismSpider** [comment]
Cancer man die form cancer

**Comment by ghost** [comment]
are you guys ready for a newer, better, oncology module for me to close?

**Comment by clusterfack** [comment]
Gotta learn some patience man

**Comment by ghost** [comment]
Everybody remembers Clor for cancer, but this guy gave us Lazarus capsules and a shit ton of viral effects as well.

**Comment by gbasood** [comment]
@cloroxygen Holy shit dude calm down with that impatience
Ask questions, and I don't just mean "who's going to merge this"
Giving and responding to feedback is more important than getting your shit merged here

**Comment by ghost** [comment]
yay

**Pull request merged**